src/renderer/VisibleRenderer.ts

```diff
@@ -4,8 +4,62 @@
  */
 import Renderer from "./Renderer";
 
+import { getFlickingAttached } from "~/utils";
+import { Panel } from "~/core";
 class VisibleRenderer extends Renderer {
+  public render() {
+    const flicking = getFlickingAttached(this._flicking, "Renderer");
+    const camera = flicking.camera;
+    const cameraElement = camera.element;
+    const visiblePanels = camera.visiblePanels;
 
+    const fragment = document.createDocumentFragment();
+
+    const visibleSortedByActualPosition = [...visiblePanels]
+      .sort((a, b) => (a.position + a.offset) - (b.position + b.offset));
+
+    const offsetGtZero = this._panels.filter(panel => panel.offset > 0);
+    const visibleOffsetLteZero = visiblePanels.filter(panel => panel.offset <= 0);
+
+    visibleSortedByActualPosition.forEach(panel => {
+      fragment.appendChild(panel.element);
+    });
+
+    // Remove remaining(removed) elements
+    while (cameraElement.firstChild) {
+      cameraElement.removeChild(cameraElement.firstChild);
+    }
+
+    cameraElement.appendChild(fragment);
+
+    const invisiblePrevPanels = this._panels.slice(0, visibleOffsetLteZero[0]?.index ?? 0);
+
+    const firstPanel = invisiblePrevPanels[0];
+    const lastPanel = invisiblePrevPanels[invisiblePrevPanels.length - 1];
+
+    // Calibrate offset created by circular
+    const firstGtPanel = offsetGtZero[0];
+    const lastGtPanel = offsetGtZero[offsetGtZero.length - 1];
+
+    const invisibleSize = this._calcPanelRangeSize(firstPanel, lastPanel)
```

Comment:
If the order of the elements(visibleSortedByActualPosition) is changed, is there any effect on invisibleSize?
If a panel with index 0 is visible, is it always 0?

Reply:
You're right, it is meant to be designed like that.